rust/ql/lib/codeql/rust/elements/ComparisonOperation.qll

```diff
@@ -0,0 +1,101 @@
+/**
+ * Provides classes for comparison operations.
+ */
+
+private import codeql.rust.elements.BinaryExpr
+private import codeql.rust.elements.Operation
+
+/**
+ * A comparison operation, such as `==`, `<` or `>=`.
+ */
+abstract private class ComparisonOperationImpl extends Operation { }
+
+final class ComparisonOperation = ComparisonOperationImpl;
+
+/**
+ * An equality comparison operation, `==` or `!=`.
+ */
+abstract private class EqualityOperationImpl extends BinaryExpr, ComparisonOperationImpl { }
+
+final class EqualityOperation = EqualityOperationImpl;
+
+/**
+ * The equal comparison operation, `==`.
+ */
+final class EqualOperation extends EqualityOperationImpl, BinaryExpr {
```

Comment:
`BinaryExpr` can be removed

Reply:
I think I prefer `EqualityOperation`.

Reply:
`EqualityOperation` is "an operation to do with equality", i.e. `==` or `!=`.

`EqualOperation` is specifically `==`.

We can't use "EqualityOperation" to describe both.

Reply:
Removed redundant `BinaryExpr`.

Reply:
OK. Then I think I slightly prefer Equal_s_Operation over EqualOperation, but I'll let you decide.

Reply:
Yeah, I agree that's a better name.  Updated.